Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ICriteriaExtensions.cs

```diff
@@ -92,5 +96,76 @@ private static string GetFullNameForView(this string viewName)
         {
             return Namespaces.Entities.NHibernate.QueryModels.GetViewNamespace(viewName);
         }
+
+        /// <summary>
+        /// Applies a join-based filter to the criteria for the specified custom authorization view.
+        /// </summary>
+        /// <param name="criteria">The criteria to which filters should be applied.</param>
+        /// <param name="viewName">The name of the view to be filtered.</param>
+        /// <param name="subjectEndpointNames">The name of the property to be joined for the entity being queried.</param>
+        /// <param name="viewTargetEndpointNames">The name of the property to be joined for the other property as authorization view.</param> 
+        /// <param name="joinType">The <see cref="JoinType" /> to be used.</param>
+        /// <param name="authViewAlias">The name of the property to be used for auth View Alias name.</param>
+        /// <param name="authorizationStrategy">The authorization strategy instance (which may be an instance providing additional context for authorization).</param>
+        public static void ApplyCustomViewJoinFilter(
+            this ICriteria criteria,
+            string viewName,
+            string[] subjectEndpointNames,
+            string[] viewTargetEndpointNames,
+            JoinType joinType,
+            string authViewAlias,
+            IAuthorizationStrategy authorizationStrategy)
+        {
+            if (authorizationStrategy is not CustomViewBasedAuthorizationStrategy customViewBasedAuthorizationStrategy)
+            {
+                throw new Exception(
+                    $"The authorization strategy supplied to the '{nameof(ApplyCustomViewJoinFilter)}' was not of the expected type '{nameof(CustomViewBasedAuthorizationStrategy)}'.");
+            }
+            
+            string authViewFullAlias = $"{CustomViewHelpers.GetAliasPrefix(viewName)}{authViewAlias}";
+
+            if (subjectEndpointNames.Length == 1)
+            {
+                // Apply authorization join using ICriteria
+                criteria.CreateEntityAlias(
+                    authViewFullAlias,
+                    Restrictions.EqProperty(
+                        $"aggregateRoot.{subjectEndpointNames[0]}",
+                        $"{authViewFullAlias}.{viewTargetEndpointNames[0]}"),
+                    joinType,
+                    customViewBasedAuthorizationStrategy.BasisEntity.EntityTypeFullName());
```

Comment:
NHibernate only supports joins to _mapped_ entities (that also don't have any mapped associations) using `ICriteria`. However, `ICriteria`, as with any other NHibernate-based query mechanism can only work with mapped entities, but given the dynamic nature of the custom views (which can be added after code generation to created corresponding entity classes and ORM mappings AND can even be added at runtime and refreshed with security metadata), we have to use some "creative" processing. For now, the query is generated to join with the "basis" entity identified based on the custom authorization view name. The generated SQL will be manipulated just prior to execution to replace the "basis" entity with the custom authorization view. See the updated NHibernate interceptor implementation.